backend/api/EventHandlers/MqttEventHandler.cs

```diff
@@ -100,15 +100,16 @@ private async void OnIsarStatus(object? sender, MqttReceivedArgs mqttArgs)
             _updateRobotSemaphore.WaitOne();
             _logger.LogDebug("Semaphore acquired for updating robot status");
 
-            var updatedRobot = await RobotService.UpdateRobotStatus(robot.Id, isarStatus.Status);
+            await RobotService.UpdateRobotStatus(robot.Id, isarStatus.Status);
+            robot.Status = isarStatus.Status;
```

Comment:
This is the main fix in the PR

Reply:
Basically, previously we were relying on the database callback to be up to date, but this tends not to be the case when calling database update. By setting the value here we make the thread execution deterministic.

Reply:
Was the updated robot not returned with the correct value?

Reply:
Assign it to a seperate value to be passed to a mission starting event along with other prerequisites